### DISCUSSION TITLE
Trying to combine arduino data with video

### DISCUSSION THREAD

**Discussion by mvustarroz** [bonsai-rx/bonsai#1344] [Q&A] [answered]
Hi, I´m quite new in Bonsai & Arduino so I'd be very grateful for any help 
I'm capturing video from my webcam and at the same time reading from the serial port of an Arduino
I'd like to add to the video the timestamp and the info coming from the Arduino
After reviewing some forums I've been able to read the info from the Arduino and to write the timestamp text on top of the video
But it seems it is impossible to combine text from Arduino + text from timestamp so I can combine both and write it on the video.
I guess it is something related to the synchr between Arduino capturing and Video capturing.
Could you help me?
Thank you very much indeed,
  Miguel
[final_test.zip](https://github.com/bonsai-rx/bonsai/files/11391297/final_test.zip)

BTW: please check the code

**Comment by bruno-f-cruz** [comment, marked as answer]
Heya,
Not sure I follow exactly what you want to achieve. Is the data from the serial synchronized with the video? i.e. is there a serial message per video frame? IF so you can use "Zip" to pair the images with the messages. If they are not aligned you must establish some sort of temporal relationship between the two streams. You can achieve this using, for example, 'WithLatestFrom' or 'CombineLatest'.
